File: PHPProject/app/Repositories/IRepository.php
<?php

namespace App\Repositories;

use Illuminate\Database\Eloquent\Model;

interface IRepository 
{
    public function getByID(int $id) : Model;
    public function create(array $data) : Model;
    public function update(int $id, array $data) : Model;
    public function deleteById(int $id): void;
}

File: NETCoreProject/NETCoreProject/Data/Interfaces/IRepository.cs
using NETCoreProject.Models;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace NETCoreProject.Data.Interfaces
{
    public interface IRepository<T> where T : class, IEntity
    {
        Task<T> GetById(int id);
        Task<T> Create(T entity);
        Task<T> Update(T entity);
        Task<T> DeleteById(int id);
    }
}
